File: Euclid/Game/Catalogue/CatalogueOrder.cs
using Euclid.Game.Entity.Humanoid;
using Euclid.Network.Streams.Util;
using Euclid.Storage.Database.Data;
using System;
using System.Collections.Generic;

namespace Euclid.Game
{
    public class CatalogueOrder : IEntity
    {
        #region Properties

        public CatalogueItem Item { get; private set; }
        public string CustomData { get; private set; }

        #endregion

        #region Constructors

        public CatalogueOrder(CatalogueItem item, string customData)
        {
            Item = item;
            CustomData = customData;
        }

        #endregion

    }
}

File: euclid/src/main/java/euclid/game/catalogue/CatalogueOrder.java
package euclid.game.catalogue;

import euclid.game.entity.IEntity;

public class CatalogueOrder implements IEntity {

    private final CatalogueItem item;
    private final String customData;

    public CatalogueOrder(CatalogueItem item, String customData) {
        this.item = item;
        this.customData = customData;
    }

    public CatalogueItem getItem() {
        return item;
    }

    public String getCustomData() {
        return customData;
    }
}
